core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -266,7 +289,8 @@ private[deploy] class Worker(
             if (registerMasterFutures != null) {
               registerMasterFutures.foreach(_.cancel(true))
             }
-            val masterAddress = masterRef.address
+            val masterAddress =
+              if (preferConfiguredMasterAddress) masterAddressToConnect.get else masterRef.address
```

Comment:
Right now `masterRef` and `masterAddressToConnect` are set at the same time. It's impossible unless we break something in future. It's better to fail rather than hiding the broken change.